src/Exception/OutOfSetException.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mathematicator\Numbers\Exception;
+
+/**
+ * Exception thrown when result cannot be provided because final number would be out of allowed set.
+ */
+class OutOfSetException extends NumberException
+{
+	public function __construct(string $haystack = '', ?string $allowed = null)
+	{
+		$allowed = $allowed ? " (allowed: $allowed)" : '';
+
+		parent::__construct("Input \"$haystack\" cannot be converted to the final set$allowed.");
```

Comment:
Ple

```suggestion
		parent::__construct('Input "' . $haystack . '" cannot be converted to the final set' . $allowed . '().');
```

Reply:
Why? It looks so dirty.

Reply:
Variable in string is magical. https://php.baraja.cz/apostrofy-a-uvozovky

Reply:
Could you please introduce me any other source that your personal webpage where is written that your notation is better? I've googled it and my notation is commonly used and secure (the same as in other languages). Btw. magic is terminus technicus, nothing like magic variable exist is PHP AFAIK.